chromatic_fitting.py

```diff
@@ -78,127 +86,277 @@ def summarize_parameters(self):
         for k, v in self.parameters.items():
             print(f"{k} =\n  {v}\n")
 
+    def reinitialize_parameters(self, exclude=[]):
+        for k, v in self.parameters.items():
+            if k not in exclude:
+                if isinstance(v, Fitted):
+                    v.clear_prior()
+
     def initialize_empty_model(self):
         """
         Restart with an empty model.
         """
-        self.model = pm.Model()
+        self.pymc3_model = pm.Model()
 
     def attach_data(self, rainbow):
         """
         Connect a `chromatic` Rainbow dataset to this object.
         """
         self.data = rainbow._create_copy()
 
+
     def white_light_curve(self):
         """
         Generate inverse-variance weighted white light curve by binning Rainbow to one bin
         """
         self.white_light = self.data.bin(nwavelengths=self.data.nwave)
 
-    def choose_optimization_method(self,optimization_method='simultaneous'):
+    def choose_optimization_method(self, optimization_method='simultaneous'):
         """
         Choose the optimization method
+        [attach_data has to be run before!]
         """
         possible_optimization_methods = ["simultaneous", "white_light", "separate"]
         if optimization_method in possible_optimization_methods:
             self.optimization = optimization_method
+            if self.optimization == "separate":
+                try:
+                    self.create_multiple_models()
+                    self.change_all_priors_to_Wavelike()
```

Comment:
This seems like a good solution for converting from a generic set of fitted/fixed parameters to ones you can use in individual separate fits, especially since you made these wrappers to be able to convert between versions.

Reply:
I think the only extra thing that might improve this solution is to save the original parameter types somewhere so if I tried to reuse the model with another optimization type then the parameters wouldn't all be wavelike